hphp/hack/test/typecheck/xhp_modifier/xhp_xhp_name.php.exp

```diff
@@ -1 +1,2 @@
-No errors
+File "xhp_xhp_name.php", line 22, characters 13-19:
+Unbound name: foo\bar (an object type) (Naming[2049])
```

Comment:
Why is this now an error? Looking at the program it's exercising:

```
xhp class foo:bar {}

xhp class foobarchildren {
  children (foo:bar)*;
}
```

Reply:
Hmm yeah this is a bit odd. The spec does allow for `xhp class foo:bar {}` which will basically be the same as:
```
namespace foo {
  xhp class bar {}
}
```
However this is not yet implemented. I attempted to add it to this PR but I ran into some issues. Since it's already quite big I think it would be better to not grow this one any further and instead take it as a follow on task. So this test does (correctly) fail, but will pass once that is implemented. For now what I'll do is remove that specific class and re-add it in the follow up.

Reply:
```
namespace foo {
  xhp class bar {}
}
```

not quite - it would be the same as:

```
xhp class foo\bar {
/ * with no change to resolution here; no new scope for `use` statements etc */
}
```

In past versions of hackc, `class foo\bar {}` was accidentally supported, with this behavior

Reply:
Ah yes, I always get this wrong. Thanks for the clarification.